src/ydata_synthetic/synthesizers/regular/cgan/model.py

```diff
@@ -124,10 +147,22 @@ def train(self, data: Union[DataFrame, array],
                 self.generator.save_weights(model_checkpoint_base_name.format('generator', epoch))
                 self.discriminator.save_weights(model_checkpoint_base_name.format('discriminator', epoch))
 
-                #Here is generating synthetic data
-                z = tf.random.normal((432, self.noise_dim))
-                label_z = tf.random.uniform((432,), minval=min(train_arguments.labels), maxval=max(train_arguments.labels)+1, dtype=tf.dtypes.int32)
-                gen_data = self.generator([z, label_z])
+                #Here is generating synthetic data from an arbitrary condition
+                gen_data = self.sample(label[0], 1000)
+
+    def sample(self, cond_array, n_samples):
+        """Produce n_samples by conditioning the generator with cond_array."""
+        assert cond_array.shape[0] == self.num_classes, \
```

Comment:
The conditional array should be a single array, with the len of the sample to be generated. 

cond_array rows = n_samples.